myriad-scheduler/src/main/java/com/ebay/myriad/api/ClustersResource.java

```diff
@@ -100,46 +95,114 @@ public Response flexUp(FlexUpClusterRequest request) {
     @Produces(MediaType.TEXT_PLAIN)
     @Consumes(MediaType.APPLICATION_JSON)
     public Response flexDown(FlexDownClusterRequest request) {
-        Preconditions.checkNotNull(request,
-                "request object cannot be null or empty");
+        Preconditions.checkNotNull(request, "request object cannot be null or empty");
 
         Integer instances = request.getInstances();
-
-        LOGGER.info("Received flexdown request");
-        LOGGER.info("Instances: " + instances);
+        String profile = request.getProfile();
+        List<String> constraints = request.getConstraints();
```

Comment:
Do we assume a single constraint for now? Do we plan more constraints? If not, why do we define it as an array?